File: app/src/main/java/beans/prj/newsapp/ui/home/HomeActivity.java
package beans.prj.newsapp.ui.home;

import beans.prj.newsapp.R;
import beans.prj.newsapp.databinding.ActivityHomeBinding;
import beans.prj.newsapp.ui.base.BaseActivity;


public class HomeActivity extends BaseActivity<ActivityHomeBinding> {

    @Override
    public int getLayoutId() {
        return R.layout.activity_home;
    }

    @Override
    public void initViewOnCreat() {
        binding.vpCategory.setAdapter(new CategoryVpAdapter(getSupportFragmentManager()));
        binding.tabLayoutCategories.setupWithViewPager(binding.vpCategory);
    }
}

File: app/src/main/java/beans/prj/newsapp/ui/home/HomeActivity.kt
package beans.prj.newsapp.ui.home

import android.os.Bundle
import androidx.databinding.DataBindingUtil
import beans.prj.newsapp.R
import beans.prj.newsapp.databinding.ActivityHomeBinding
import beans.prj.newsapp.ui.base.BaseActivity

class HomeActivity : BaseActivity<ActivityHomeBinding>() {

    override lateinit var binding: ActivityHomeBinding



    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = DataBindingUtil.setContentView(this, R.layout.activity_home)

        binding.apply {
            vpCategory.adapter = CategoryVpAdapter(supportFragmentManager)
            tabLayoutCategories.setupWithViewPager(vpCategory)

            executePendingBindings()
        }


    }
}